File: Reservation.ts
import { RentalItem } from "./RentalItem";

export class Reservation {
  item: RentalItem;
  startDate: Date;
  endDate: Date;

  constructor(item: RentalItem, startDate: Date, days: number) {
    this.item = item;
    this.startDate = startDate;
    this.endDate = new Date(startDate);
    this.endDate.setDate(startDate.getDate() + days);
  }

  toString(): string {
    return `${this.item.name} réservé du ${this.startDate.toDateString()} au ${this.endDate.toDateString()}`;
  }
}

File: Reservation.java
import java.time.LocalDate;

public class Reservation {
    private RentalItem item;
    private LocalDate dateDebut;
    private LocalDate dateFin;

    public Reservation(RentalItem item, LocalDate dateDebut, int jours) {
        this.item = item;
        this.dateDebut = dateDebut;
        this.dateFin = dateDebut.plusDays(jours);
    }

    public RentalItem getItem() {
        return item;
    }

    public LocalDate getDateDebut() {
        return dateDebut;
    }

    public LocalDate getDateFin() {
        return dateFin;
    }

    @Override
    public String toString() {
        return item.getNom() + " réservé du " + dateDebut + " au " + dateFin;
    }
}
